src/App.tsx

```diff
@@ -1,26 +1,363 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import {
+  deleteTodo,
+  getTodos,
+  postTodo,
+  updateTodo,
+  USER_ID,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import { Footer } from './components/Footer';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { ErrorMessages } from './types/ErrorMessages';
+import { Filter } from './types/Filter';
+import classNames from 'classnames';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [input, setInput] = useState('');
+  const [errorMessage, setErrorMessage] = useState('');
+  const [filter, setFilter] = useState<Filter>(Filter.ALL);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loadingTodoIds, setLoadingTodoIds] = useState<number[]>([]);
+  const [editingTodoId, setEditingTodoId] = useState<number | null>(null);
+  const [editLoader, setEditLoader] = useState<number | null>(null);
```

Comment:
You could use the same `loadingTodoIds` instead - i.e. on edit add corresponding todo id to the array